sync/sync_head.go

```diff
@@ -186,7 +192,59 @@ func (s *Syncer[H]) verify(ctx context.Context, newHead H) (bool, error) {
 			"reason", verErr.Reason)
 	}
 
-	return verErr.SoftFailure, err
+	return false, err
+}
+
+// verifySkipping performs a bifurcated header verification process such that
+// it tries to find a header (or several headers if necessary)
+// between the networkHead and the subjectiveHead such that non-adjacent
+// (or in the worst case adjacent) verification passes and the networkHead
+// can be verified as a valid sync target against the syncer's subjectiveHead.
+//
+// When such candidates cannot be found [NewValidatorSetCantBeTrustedError] will be returned.
+func (s *Syncer[H]) verifySkipping(ctx context.Context, subjHead, networkHeader H) error {
+	subjHeight := subjHead.Height()
+
+	diff := networkHeader.Height() - subjHeight
+	if diff <= 0 {
+		panic(fmt.Sprintf("implementation bug:\n subjective head height %d, hash %X,\n network head height %d, hash %X",
+			subjHeight, subjHead.Hash(),
+			networkHeader.Height(), networkHeader.Hash(),
+		))
+	}
```

Comment:
Do you think we need this? As this method is called only after preliminary `Verify` it's really impossible to have this in practice.

Reply:
Well, that's why it's a panic and not an error. I don't have a strong argument to keep it but not going argue why it should be here. If you think this method will be simpler without it - I will drop this panic.

Reply:
Dropped after all.